src/main/java/org/opensearch/flowframework/common/DefaultUseCases.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.common;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Enum encapsulating the different default use cases and templates we have stored
+ */
+public enum DefaultUseCases {
+
+    /** defaults file and substitution ready template for OpenAI embedding model */
+    OPEN_AI_EMBEDDING_MODEL_DEPLOY(
+        "open_ai_embedding_model_deploy",
+        "defaults/open-ai-embedding-defaults.json",
+        "substitutionTemplates/deploy-remote-model-template.json"
+    ),
+    /** defaults file and substitution ready template for cohere embedding model */
+    COHERE_EMBEDDING_MODEL_DEPLOY(
+        "cohere-embedding_model_deploy",
+        "defaults/cohere-embedding-defaults.json",
+        "substitutionTemplates/deploy-remote-model-template-extra-params.json"
+    ),
+    /** defaults file and substitution ready template for local neural sparse model and ingest pipeline*/
+    LOCAL_NEURAL_SPARSE_SEARCH(
+        "local_neural_sparse_search",
+        "defaults/local-sparse-search-defaults.json",
+        "substitutionTemplates/neural-sparse-local-template.json"
+    );
+
+    private final String useCaseName;
+    private final String defaultsFile;
+    private final String substitutionReadyFile;
+    private static final Logger logger = LogManager.getLogger(DefaultUseCases.class);
+    private static final Set<String> allResources = Stream.of(values()).map(DefaultUseCases::getDefaultsFile).collect(Collectors.toSet());
```

Comment:
Not using `allResources` anywhere?

If want to use somewhere, use `EnumSet.allOf(DefaultUseCases.class)` instead of `values` to avoid issues with enum order.

```java
EnumSet.allOf(DefaultUseCases.class).stream()
            .map(DefaultUseCases::getDefaultsFile)
            .collect(Collectors.toSet());
```